src/palette/internal/palettecompat.cpp

```diff
@@ -69,3 +73,28 @@ void PaletteCompat::migrateOldPaletteItemIfNeeded(ElementPtr& element, Score* pa
         element.reset(newExpression);
     }
 }
+
+void PaletteCompat::addNewItemsIfNeeded(Palette& palette, Score* paletteScore)

```

Comment:
Is there no version information that we can use to determine whether the palette has never been updated yet? Is that the cause of the "Known issue" you mentioned, or is that something different?